.pipelines/UpdateVersions.ps1

```diff
@@ -0,0 +1,126 @@
+        $newVersionString = 'package id="Microsoft.WindowsAppSDK" version="' + $WinAppSDKVersion + '"'
+        $oldVersionString = 'package id="Microsoft.WindowsAppSDK" version="[-.0-9a-zA-Z]*"'
+        $content = $content -replace $oldVersionString, $newVersionString
+        Set-Content -Path $_.FullName -Value $content
+        Write-Host "Modified " $_.FullName 
+    }
+}
+
+# Update Directory.Packages.props file
+$propsFile = "Directory.Packages.props"
+if (Test-Path $propsFile) {
+    $content = Get-Content $propsFile -Raw
+    if ($content -match '<PackageVersion Include="Microsoft.WindowsAppSDK"') {
+        $newVersionString = '<PackageVersion Include="Microsoft.WindowsAppSDK" Version="' + $WinAppSDKVersion + '" />'
+        $oldVersionString = '<PackageVersion Include="Microsoft.WindowsAppSDK" Version="[-.0-9a-zA-Z]*" />'
+        $content = $content -replace $oldVersionString, $newVersionString
+        Set-Content -Path $propsFile -Value $content
+        Write-Host "Modified " $propsFile
+    }
+}
+
+# Update .vcxproj files
+Get-ChildItem -Recurse *.vcxproj | ForEach-Object {
+    $content = Get-Content $_.FullName -Raw
+    if ($content -match '\\Microsoft.WindowsAppSDK.') {
+        $newVersionString = '\Microsoft.WindowsAppSDK.' + $WinAppSDKVersion + '\'
+        $oldVersionString = '\\Microsoft.WindowsAppSDK.[-.0-9a-zA-Z]*\\'
+        $content = $content -replace $oldVersionString, $newVersionString
+        Set-Content -Path $_.FullName -Value $content
+        Write-Host "Modified " $_.FullName
+    }
+}
+
+# Update .csproj files
+Get-ChildItem -Recurse *.csproj | ForEach-Object {
+    $content = Get-Content $_.FullName -Raw
+    if ($content -match 'PackageReference Include="Microsoft.WindowsAppSDK"') {
+        $newVersionString = 'PackageReference Include="Microsoft.WindowsAppSDK" Version="'+ $WinAppSDKVersion + '"'
+        $oldVersionString = 'PackageReference Include="Microsoft.WindowsAppSDK" Version="[-.0-9a-zA-Z]*"'
+        $content = $content -replace $oldVersionString, $newVersionString
+        Set-Content -Path $_.FullName -Value $content
+        Write-Host "Modified " $_.FullName 
+    }
+}
+
+# Load the nuget.config file
+Write-Host "Updating nuget.config file"
+$filePath = "nuget.config"
+[xml]$xml = Get-Content -Path $filePath
+
+# Add localpackages source into nuget.config
+$packageSourcesNode = $xml.configuration.packageSources
+$addNode = $xml.CreateElement("add")
+$addNode.SetAttribute("key", "localpackages")
+$addNode.SetAttribute("value", "localpackages")
+$packageSourcesNode.AppendChild($addNode) | Out-Null
+
+# Remove <packageSourceMapping> tag and its content
+$packageSourceMappingNode = $xml.configuration.packageSourceMapping
+if ($packageSourceMappingNode) {
+    $xml.configuration.RemoveChild($packageSourceMappingNode) | Out-Null
+}
+
+# print nuget.config after modification
+$xml.OuterXml
+# Save the modified nuget.config file
+$xml.Save($filePath)
```

Comment:
For these big chunk of code, would you mind to break it down into smaller functions?
As the result, the main one will looks cleaner

Reply:
e.g.
function Update-PackageConfig {
    param (
        [string]$filePath,
        [string]$oldVersionString,
        [string]$newVersionString
    )
    # Function implementation
}